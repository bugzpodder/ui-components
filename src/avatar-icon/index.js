// @flow
export { AvatarIcon } from "./avatar-icon";
