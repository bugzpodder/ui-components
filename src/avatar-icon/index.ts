export { AvatarIcon } from "./avatar-icon";
