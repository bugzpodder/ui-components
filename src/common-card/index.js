// @flow
export { CommonCard } from "./card";
