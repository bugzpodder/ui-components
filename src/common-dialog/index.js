//@flow
export { CommonDialog } from "./dialog";
