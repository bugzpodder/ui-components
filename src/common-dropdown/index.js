// @flow
export { CommonDropdownMenu } from "./dropdown-menu";
