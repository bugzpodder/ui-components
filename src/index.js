// @flow
import type { PagedTableProps } from "./table";

export { Alert } from "./alert";
export { AvatarIcon } from "./avatar-icon";
export { CommonCard } from "./common-card";
export { BreastCancerRibbon } from "./logos";
export { CommonDialog, CommonMultiPageDialog } from "./common-dialog";
export { CommonDropdownMenu } from "./common-dropdown";
export { CommonPanel } from "./common-panel";
export { CommonMultiSelect, CommonSelect } from "./common-select";
export { CommonSuggest, SuggestionsOmniField } from "./common-suggest";
export { CommonSwitch } from "./common-switch";
export {
  CommonPage, CommonTabbedPage, CommonPageV2, CommonTabbedPageV2,
} from "./page";
export {
  DateValue,
  DateTimeValue,
  HumanizedDateTime,
  DateInput,
  DateTimeInput,
  wrapPickerUtilProvider,
  TimeInput,
} from "./date";
export { ExternalLink, ExternalLinkButton, LinkButton } from "./link";
export { Navbar } from "./__deprecated__/navbar";
export { NavbarV2 } from "./navbar";
export { NotificationCenter } from "./notification-center";
export { ReadOnlyTextField, ValidatedReadOnlyTextField } from "./readonly-text-field";
export { OmniChips, OmniSearchBar } from "./omni-search-bar";
export {
  SimpleTable, ExportTableButton, LargeSimpleTable, PagedTable,
} from "./table";
export { ExportButton, generateReport } from "./export-button";
export type { PagedTableProps };
export { SignInPage } from "./sign-in-page";
export { SpinnerOverlay } from "./spinner-overlay";
export { StyleWrapper } from "./style-wrapper";
export { TabbedTimelineCard, TimelineGraph, TimelineCard } from "./timeline-graph";
export { TextInput, TextInputV2 } from "./text-input";
export { TwoColumnGrid, TwoColumnRow } from "./two-column-grid";
export { UploadButton } from "./upload-button";
export { SelectionGrid } from "./selection-grid";
export {
  // $FlowFixMe: Property `readFileListAsTextArray` is missing in ModuleNamespace for export {}
  readFileListAsTextArray,
  // $FlowFixMe: Property `readFileListAsText` is missing in ModuleNamespace for export {}
  readFileListAsText,
  // $FlowFixMe: Property `readFileAsText` is missing in ModuleNamespace for export {}
  readFileAsText,
} from "./utils";
