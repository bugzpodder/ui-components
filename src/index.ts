export * from "./alert";
export * from "./avatar-icon";
export * from "./common-card";
export { BreastCancerRibbon } from "./logos";
export { CommonDialog, CommonMultiPageDialog } from "./common-dialog";
export { CommonDropdownMenu } from "./common-dropdown";
export * from "./common-panel";
export * from "./common-select";
export * from "./common-suggest";
export * from "./common-switch";
export * from "./page";
export * from "./date";
export * from "./link";
export { Navbar } from "./__deprecated__/navbar";
export * from "./navbar";
export * from "./notification-center";
export * from "./readonly-text-field";
export * from "./omni-search-bar";
export * from "./table";
export * from "./export-button";
export * from "./sign-in-page";
export * from "./spinner-overlay";
export * from "./style-wrapper";
export * from "./timeline-graph";
export * from "./text-input";
export * from "./two-column-grid";
export * from "./upload-button";
export * from "./selection-grid";
export * from "./utils";
export * from "./types/base-navbar";
export * from "./types/dropdown";
export * from "./types/paged-table";
export * from "./types/selection-grid";
export * from "./types/timeline";
export * from "./types/card";
export * from "./types/grid";
export * from "./types/panel";
export * from "./types/switch";
export * from "./types/dialog";
export * from "./types/notification";
export * from "./types/select";
export * from "./types/table";
export { FavoriteIconButton } from "./dev";
