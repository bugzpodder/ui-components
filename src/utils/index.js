// @flow
export { COUNTRIES, SERGEY } from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
