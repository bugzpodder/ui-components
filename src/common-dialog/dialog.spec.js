// @flow
import React from "react";
import { renderIntoDocument, cleanup, Simulate } from "react-testing-library";
import { bindElementToQueries } from "dom-testing-library";
import { TestWrapper } from "@grail/components/src/utils";
import { CommonDialog, CommonMultiPageDialog } from "./index";

const bodyUtils = bindElementToQueries(document.body);

test("render dialog", () => {
	const mockCallback = jest.fn();
	renderIntoDocument(
		<TestWrapper>
			<CommonDialog
				actions={[
					{ id: "action-button", name: "action one", callback: jest.fn() },
					{ name: "action two", callback: jest.fn() },
				]}
				title="test"
				isVisible={true}
				hideModal={mockCallback}
			>
				Test Content
			</CommonDialog>
		</TestWrapper>,
	);
	Simulate.click(bodyUtils.getByText("Close"));
	expect(mockCallback).toBeCalled();
	expect(document.body).toMatchSnapshot();
	cleanup();
});

test("render multi page dialog", () => {
	const mockCallback = jest.fn();
	renderIntoDocument(
		<TestWrapper>
			<CommonMultiPageDialog
				actions={[
					{ id: "action-button", name: "action one", callback: jest.fn(), pages: [0] },
					{ id: "action-button", name: "action two", callback: jest.fn(), pages: [1] },
				]}
				title="test"
				isVisible={true}
				hideModal={jest.fn()}
				pages={[
					<div
						key="0"
						data-testid="page">
						Page 1
					</div>,
					<div
						key="1"
						data-testid="page">
						Page 2
					</div>,
				]}
				pageIndex={1}
				setPage={mockCallback}
			>
				Test Content
			</CommonMultiPageDialog>
		</TestWrapper>,
	);
	Simulate.click(bodyUtils.getByText("Back"));
	expect(mockCallback).toBeCalled();
	expect(document.body).toMatchSnapshot();
	cleanup();
});
