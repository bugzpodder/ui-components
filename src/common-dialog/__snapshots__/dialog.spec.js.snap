// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render dialog 1`] = `
<body
  style="overflow: hidden; padding-right: 0px;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root MuiDialog-root commonDialog"
    role="dialog"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation24 MuiPaper-rounded commonDialogPaper"
      role="document"
      style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiDialogTitle-root"
        id="form-dialog-title"
      >
        <h2
          class="MuiTypography-root MuiTypography-title"
        >
          test
        </h2>
      </div>
      <div
        class="MuiDialogContent-root commonDialogContent"
      >
        Test Content
      </div>
      <div
        class="MuiDialogActions-root"
      >
        <div
          class="commonDialogFooter MuiDialogActions-action"
        >
          <div
            class="left"
          />
          <div
            class="right"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-flat"
              id="close"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Close
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-textPrimary MuiButton-flat MuiButton-flatPrimary"
              data-id="action-button"
              data-is-enabled="true"
              data-testid="action-button"
              id="action-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                action one
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-textPrimary MuiButton-flat MuiButton-flatPrimary"
              data-is-enabled="true"
              id="action two"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                action two
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`render multi page dialog 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root MuiDialog-root commonDialog"
    role="dialog"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation24 MuiPaper-rounded commonDialogPaper"
      role="document"
      style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiDialogTitle-root"
        id="form-dialog-title"
      >
        <h2
          class="MuiTypography-root MuiTypography-title"
        >
          test
        </h2>
      </div>
      <div
        class="MuiDialogContent-root commonDialogContent"
      >
        <div
          data-testid="page"
        >
          Page 2
        </div>
      </div>
      <div
        class="MuiDialogActions-root"
      >
        <div
          class="commonDialogFooter MuiDialogActions-action"
        >
          <div
            class="left"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-flat back"
              id="back"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root margin-right-5"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    />
                  </g>
                </svg>
                 Back
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="right"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-flat"
              id="close"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Close
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-textPrimary MuiButton-flat MuiButton-flatPrimary"
              data-id="action-button"
              data-is-enabled="true"
              data-testid="action-button"
              id="action-button"
              pages="1"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                action two
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
